wgpu/backends/wgpu_native/_api.py

```diff
@@ -2160,101 +2183,92 @@ def set_bind_group(
         offsets = list(dynamic_offsets_data)
         c_offsets = ffi.new("uint32_t []", offsets)
         bind_group_id = bind_group._internal
-        if isinstance(self, GPUComputePassEncoder):
-            # H: void f(WGPUComputePassEncoder computePassEncoder, uint32_t groupIndex, WGPUBindGroup group, size_t dynamicOffsetCount, uint32_t const * dynamicOffsets)
-            libf.wgpuComputePassEncoderSetBindGroup(
-                self._internal, index, bind_group_id, len(offsets), c_offsets
-            )
-        else:
-            # H: void f(WGPURenderPassEncoder renderPassEncoder, uint32_t groupIndex, WGPUBindGroup group, size_t dynamicOffsetCount, uint32_t const * dynamicOffsets)
-            libf.wgpuRenderPassEncoderSetBindGroup(
-                self._internal,
-                index,
-                bind_group_id,
-                len(offsets),
-                c_offsets,
-            )
+
+        # We will call one of
+        # H: void f(WGPUComputePassEncoder computePassEncoder, uint32_t groupIndex, WGPUBindGroup group, size_t dynamicOffsetCount, uint32_t const * dynamicOffsets)
+        # libf.wgpuComputePassEncoderSetBindGroup
+        # H: void f(WGPURenderPassEncoder renderPassEncoder, uint32_t groupIndex, WGPUBindGroup group, size_t dynamicOffsetCount, uint32_t const * dynamicOffsets)
+        # libf.wgpuRenderPassEncoderSetBindGroup
+        # H: void f(WGPURenderBundleEncoder renderBundleEncoder, uint32_t groupIndex, WGPUBindGroup group, size_t dynamicOffsetCount, uint32_t const * dynamicOffsets)
+        # libf.wgpuRenderBundleEncoderSetBindGroup
+        type(self).set_bind_group_function(
+            self._internal, index, bind_group_id, len(offsets), c_offsets
+        )
 
 
 class GPUDebugCommandsMixin(classes.GPUDebugCommandsMixin):
     def push_debug_group(self, group_label):
         c_group_label = ffi.new("char []", group_label.encode())
-        color = 0
-        # todo: these functions are temporarily not available in wgpu-native
-        return  # noqa
-        if isinstance(self, GPUComputePassEncoder):
-            # H: void f(WGPUComputePassEncoder computePassEncoder, char const * groupLabel)
-            libf.wgpuComputePassEncoderPushDebugGroup(
-                self._internal, c_group_label, color
-            )
-        else:
-            # H: void f(WGPURenderPassEncoder renderPassEncoder, char const * groupLabel)
-            libf.wgpuRenderPassEncoderPushDebugGroup(
-                self._internal, c_group_label, color
-            )
+        function = type(self).push_debug_group_function
```

Comment:
For the record, getting the type should not be necessary, this should work as well.
```py
        function = self.push_debug_group_function
```

Though it could be that your version is faster, because it won't have to check `self.__dict__` first, so we can leave it as-is.

Reply:
Moot comment once we move to `self.__pick_render_function()`.

Reply:
I'm glad to see that I'm not the only one who was surprised that I couldn't write this.  I did originally write `self.push_debug_group_function`, but this doesn't work!.  

Here's a simpler code snippet.

```
def global_printer(a):
    print("Printer:", a)

class MyClass:
    printer = global_printer

    def print_it(self, object):
        function = self.printer
        function(object)

MyClass().print_it("Hello")
```
which when run gives the strange error message.
```
TypeError: global_printer() takes 1 positional argument but 2 were given
```
When Python sees that that value of  `<instance>`.`<field>` is an unbound function, it binds `<instance>` to the first argument.

I have to write either `self.__class__` or `type(self).` to force Python to access the field without binding the first argument.  The latter looked a little bit nicer.

Reply:
The latter also has better portability guarantees